### PR TITLE
 fix: allow gcloud installation on run tests

### LINKED ISSUE #735: rspec: E2E tests Can make request to service failed
This test failed!

To configure my behavior, see [the Flaky Bot documentation](https://github.com/googleapis/repo-automation-bots/tree/master/packages/flakybot).

If I'm commenting on this issue too often, add the `flakybot: quiet` label and
I will stop commenting.

---

commit: 0dc6e055ae9969d5aeb1e5de25a3dad588cf0cc4
buildURL: [Build Status](https://source.cloud.google.com/results/invocations/d15146fa-e7cc-494b-aecb-8d5dada226cb), [Sponge](http://sponge2/d15146fa-e7cc-494b-aecb-8d5dada226cb)
status: failed
<details><summary>Test output</summary><br><pre>Failure/Error: raise "Error: No service url found. For example: https://service-x8xabcdefg-uc.a.run.app"

RuntimeError:
  Error: No service url found. For example: https://service-x8xabcdefg-uc.a.run.app
./spec/e2e_spec.rb:50:in `block (2 levels) in <top (required)>'</pre></details>

### PR DESCRIPTION
**Pull request by averikitsch**
## Description

Fixes #735

Note: It's a good idea to open an issue first for discussion.

## Checklist
- [x] **Tests** pass
- [x] **Lint** pass: `bundle exec rubocop`
- [x] Please **merge** this PR for me once it is approved.

### CODE REVIEW

**Comment by averikitsch** [comment]
Test failures are from pub/sub sample which is also failing on continuous testing.

File: eventarc/generic/Dockerfile [hunk 1]
```diff
@@ -16,7 +16,7 @@
 
 # Use the official lightweight Ruby image.
 # https://hub.docker.com/_/ruby
-FROM ruby:2.7-slim
+FROM ruby:2.5-slim
```
Comment (dazuma) [inline eventarc/generic/Dockerfile:19]: I don't think we want to downgrade to Ruby 2.5. Especially as this is the image we're deploying to the Cloud Run service, not the Ruby that's running the tests. For a customer-visible sample, better to show a current Ruby.

Same with the pubsub image below.

File: eventarc/generic/Gemfile.lock [hunk 2]
```diff
@@ -62,4 +62,4 @@ DEPENDENCIES
   sinatra (~> 2.0)
 
 BUNDLED WITH
-   2.1.4
+  1.17.3
```
Comment (dazuma) [inline eventarc/generic/Gemfile.lock:65]: Similarly, we really should stay with bundler 2 to build the runtime image.

Same with the pubsub sample below.

**Review by dazuma — CHANGES_REQUESTED** [review]

File: eventarc/generic/Gemfile.lock (hunk 2 above)
Comment (averikitsch) [reply eventarc/generic/Gemfile.lock:65]: I've had to downgrade elsewhere too (https://github.com/GoogleCloudPlatform/ruby-docs-samples/pull/716) due to the test infra setup. Without this the tests fail and stop. The dockerfile needs to be downgrade to work with this bundler version from my understanding.

Comment (dazuma) [reply eventarc/generic/Gemfile.lock]: Yes, our test infra right now is somewhat broken in that it doesn't have bundler 2 installed, so for _most_ tests, Gemfiles need to reference bundler 1. However, I think _these_ bundles are not installed by the test infra, but are part of the images that get deployed to cloud run (i.e. the bundle install happens inside the Dockerfile.) So I _think_ we should be able to use an up-to-date bundler and an up-to-date Ruby.

Comment (averikitsch) [reply eventarc/generic/Gemfile.lock]: I've reverted the changes, but see an error:
```
bundle exec ruby /tmpfs/src/github/ruby-docs-samples/spec/e2e_cleanup.rb eventarc/generic 1a33ed0017
/usr/local/lib/ruby/site_ruby/2.4.0/bundler/lockfile_parser.rb:108:in `warn_for_outdated_bundler_version': You must use Bundler 2 or greater with this lockfile. (Bundler::LockfileError)
```

Comment (dazuma) [reply eventarc/generic/Gemfile.lock]: Ugh. Looks like e2e_cleanup is doing the wrong thing; for one thing, it's assuming that e2e means app engine (which it clearly doesn't in this case). And it shouldn't be trying to install the bundle in those directories—that's a useless exercise. I suspect a long-term fix is going to require much deeper surgery into the test scripts.

Maybe for now it's okay to switch the bundler and Ruby versions to deal with the immediate test failures, much as it makes me sad.

Comment (averikitsch) [reply eventarc/generic/Gemfile.lock:65]: I've reverted the changes. Thanks for looking at this PR. I will add these notes to the sample testing WG sync.

Comment (dazuma) [reply eventarc/generic/Gemfile.lock:65]: Thanks Averi. Sorry about the churn on this. Graham and I will need to take a closer look at all this and get it cleaned up. The rest looks fine in the meantime, so LGTM.

**Review by dazuma — APPROVED** [review]

**Pull request merged**